packages/drivers/odsp-driver/src/test/odspDriverResolverTest.spec.ts

```diff
@@ -64,16 +66,120 @@ describe("Odsp Driver Resolver", () => {
 
         const [, queryString] = request.url.split("?");
         const searchParams = new URLSearchParams(queryString);
-        assert.equal(searchParams.get("path"), filePath, "filePath should match");
-        assert.equal(searchParams.get("driveId"), driveId, "Drive id should match");
+        assert.strictEqual(searchParams.get("path"), filePath, "filePath should match");
+        assert.strictEqual(searchParams.get("driveId"), driveId, "Drive id should match");
+    });
+
+    it("Non-Empty string filePath is resolvable", async () => {
+        // Arrange
+        filePath = "data1";
+        const itemId = "item1";
+        const testRequest: IRequest = {
+            url: `${siteUrl}?driveId=${driveId}&path=${filePath}&itemId=${itemId}`,
+            headers: { createNew:{ fileName:`${fileName}` } },
+        };
+
+        // Act
+        const resolvedUrl = await resolver.resolve(testRequest);
+        ensureFluidResolvedUrl(resolvedUrl);
+
+        // Assert
+        assert.strictEqual(resolvedUrl.fileName, fileName, "FileName should be equal");
+        assert.strictEqual(resolvedUrl.driveId, driveId, "Drive id should be equal");
+        assert.strictEqual(resolvedUrl.siteUrl, siteUrl, "SiteUrl should be equal");
+        assert.strictEqual(resolvedUrl.itemId, "", "Item id should be absent");
+        assert.strictEqual(resolvedUrl.hashedDocumentId, "", "No doc id should be present");
+        assert.strictEqual(resolvedUrl.endpoints.snapshotStorageUrl, "", "Snapshot url should be empty");
+
+        const expectedResolvedUrl = `fluid-odsp://${siteUrl}?driveId=${driveId}&path=${filePath}&itemId=${itemId}`
+        + `&version=null`;
+        assert.strictEqual(resolvedUrl.url, expectedResolvedUrl, "resolved url is wrong");
+    });
+
+    it("FilePath with 3 data object ids is resolvable", async () => {
+        // Arrange
+        filePath = "data1/data2/data3";
+        const itemId = "item1";
+        const testRequest: IRequest = {
+            url: `${siteUrl}?driveId=${driveId}&path=${filePath}&itemId=${itemId}`,
+        };
+
+        // Act
+        const resolvedUrl = await resolver.resolve(testRequest);
+        ensureFluidResolvedUrl(resolvedUrl);
+
+        // Assert
+        assert.strictEqual(resolvedUrl.fileName, "", "FileName should be absent");
+        assert.strictEqual(resolvedUrl.driveId, driveId, "Drive id should be equal");
+        assert.strictEqual(resolvedUrl.siteUrl, siteUrl, "SiteUrl should be equal");
+        assert.strictEqual(resolvedUrl.itemId, itemId, "Item id should be equal");
+        assert.strictEqual(resolvedUrl.hashedDocumentId, getHashedDocumentId(driveId, itemId),
+            "Doc id should be present");
+        assert.notStrictEqual(resolvedUrl.endpoints.snapshotStorageUrl, "", "Snapshot url should be present");
+
+        const expectedResolvedUrl = `fluid-odsp://placeholder/placeholder/${resolvedUrl.hashedDocumentId}/${filePath}`
```

Comment:
Not related to this PR - I didn't realize that the osdpDriverUrlResolver returns a url with the word placeholder in it! That's so weird :-).